presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java

```diff
@@ -395,62 +392,60 @@ protected Boolean visitQuerySpecification(QuerySpecification node, Integer level
         @Override
         protected Boolean visitQuery(Query node, Integer level)
         {
-            if (node.getWith().isPresent()) {
-                With with = node.getWith().get();
-                for (WithQuery query : with.getQueries()) {
-                    if (process(query.getQuery(), level + 1)) {
-                        return true;
-                    }
-                }
+            if (node.getChildren().stream().anyMatch(n -> process(n, level))) {
+                return true;
```

Comment:
One thing to note is `getChildren` is not precomputed and could be a costly call. The implementation from Query.java is something like this
```
  public List<Node> getChildren()
    {
        ImmutableList.Builder<Node> nodes = ImmutableList.builder();
        with.ifPresent(nodes::add);
        nodes.add(queryBody);
        orderBy.ifPresent(nodes::add);
        offset.ifPresent(nodes::add);
        return nodes.build();
    }
 ```
 I think we are only interested in the with clause here and queryBody here ?

Reply:
This is true for all the classes. All of the classes create new immutable lists on each invocation. We can refactor those to eagerly create children list but that would be a larger refactor + need to consider the memory impact on the critical query execution path.